hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1935,6 +1938,21 @@ public String listStatus(final Path path, final String startFrom,
       } while (shouldContinue);
 
       fileStatuses.addAll(fileMetadata.values());
+
+      if (fileStatuses.size() == 0) {
+        FileStatus status = getFileStatus(path, tracingContext, true);
+        if (status.isFile()) {
+          fileStatuses.add(status);
+          return continuation;
+        }
+      }
+
+      LOG.debug("List Status on Blob Endpoint on filesystem: {} path: {} received {} objects from server",
+          client.getFileSystem(), path, objectCountReturnedByServer);
```

Comment:
maybe add returned to user as well to same log line ?

Reply:
Taken